capsules/src/usb/usbc_client_ctrl.rs

```diff
@@ -274,56 +246,25 @@ impl<'a, 'b, C: hil::usb::UsbController<'a>> ClientCtrl<'a, 'b, C> {
                     DescriptorType::Device => match descriptor_index {
                         0 => {
                             let buf = self.descriptor_buf();
-                            let len = self.device_descriptor.write_to(buf);
+                            let len = self.device_descriptor_buffer.write_to(buf);
 
                             let end = min(len, requested_length as usize);
+
                             self.state[endpoint].set(State::CtrlIn(0, end));
                             hil::usb::CtrlSetupResult::Ok
                         }
                         _ => hil::usb::CtrlSetupResult::ErrInvalidDeviceIndex,
                     },
-                    DescriptorType::Configuration => {
-                        match descriptor_index {
-                            0 => {
-                                // Place all the descriptors related to this configuration into a
-                                // buffer contiguously.
-
-                                let buf = self.descriptor_buf();
-                                let mut len = 0;
-
-                                // A single configuration, with the following interface.
-                                len += self.configuration_descriptor.write_to(&buf[len..]);
-
-                                // A single interface, with the following descriptors and endpoints.
-                                len += self.interface_descriptor.write_to(&buf[len..]);
-
-                                // HID descriptor, if any.
-                                if let Some(dh) = self.hid_descriptor {
-                                    len += dh.write_to(&buf[len..]);
-                                }
-
-                                // Endpoints.
-                                for de in self.endpoint_descriptors {
-                                    len += de.write_to(&buf[len..]);
-                                }
-
-                                let end = min(len, requested_length as usize);
-                                self.state[endpoint].set(State::CtrlIn(0, end));
-                                hil::usb::CtrlSetupResult::Ok
-                            }
-                            _ => hil::usb::CtrlSetupResult::ErrInvalidConfigurationIndex,
-                        }
-                    }
-                    DescriptorType::Interface => match descriptor_index {
```

Comment:
I haven't tested it. But does anything break if a host sends us a query for an Interface descriptor anyway?

Reply:
Interface descriptor requests are not possible. Per multiple resources, GET_DESCRIPTOR cannot be used to access interfaces or endpoints separately from configurations. https://www.beyondlogic.org/usbnutshell/usb6.shtml So it should be fine to hit the default match case.